packages/contractkit/src/wrappers/MultiSig.ts

```diff
@@ -0,0 +1,28 @@
+import { Address } from '../base'
+import { MultiSig } from '../generated/types/MultiSig'
+import { BaseWrapper, proxyCall, proxySend, stringToBytes } from './BaseWrapper'
+
+export type SubmitTransactionsParams = Parameters<MultiSig['methods']['submitTransaction']>
+export const submitTransactionsParams = (
+  destination: Address,
+  transaction: string
+): SubmitTransactionsParams => {
+  return [destination, 0, stringToBytes(transaction)]
+}
+
+/**
+ * Contract for handling multisig actions
+ */
+export class MultiSigWrapper extends BaseWrapper<MultiSig> {
+  /**
+   * Allows an owner to submit and confirm a transaction.
+   * @param index The index of the pending withdrawal to withdraw.
+   */
+  submitTransaction = proxySend(
```

Comment:
What if this function took a `TransactionObject`? That way we could leave the existing wrappers as-is (i.e. multisig agnostic, which I think is preferable), and users would be able to submit transactions more generally, leveraging the other contractkit wrappers, e.g.

```
const governance = await kit.getGovernanceWrapper()
const approverMultisig = await kit.getMultisigWrapperAt('0x1234567890...')
const tx = await governance.approve(10)
await approverMultisig.submitOrConfirmTransaction(tx)
```

Reply:
> we could leave the existing wrappers as-is
^ are you suggesting we have the caller (test, cli, contractkit users) reference the multisig itself?  Is this a common pattern for multisig?  I feel it makes sense to imbed the multisig reference in the Governance and Reserve wrappers to abstract some of the complexity -- the caller would not have worry about how its implemented (multisig or not) and would not make the mistake of ever trying to perform the action forgetting it needs to be sent through the multisig contract.

i suppose the negative here is that the spender/approver's address could be changed to not use multisig (so multisig is not permanently inherent to the smart contract implementations of these roles).

Reply:
i guess...how would the client know they need to submit request through a multisig contract?

i suppose we also lose clarity that the actions may not actually be performed by rather submitted, awaiting to be confirmed by another signatory.  so the method name might need to be something like submitOrCommitGoldTransfer

Reply:
Sorry for the drive-by comment here, but my interpretation of Asa's suggestion was to leave the Governance/Reserve wrappers indeed independent of the multisig itself. The tooling/client in this case would be aware that it is being called through a multisig.

> i suppose the negative here is that the spender/approver's address could be changed to not use multisig (so multisig is not permanently inherent to the smart contract implementations of these roles).

How is that a problem? That actually seems like the benefit of not coupling the multisig to the Governance/Reserve wrappers?

Reply:
its not a problem -- that was me arguing in favor of asa's suggestion

i just threw out pros/cons without much clarity for which is which